dali/python/nvidia/dali/numba_func.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from numba import types
+
+def _populate_setup_args(types_list, is_input_args=False):
+    ret = []
+    for type in types_list:
+        ret.append(types.CPointer(type))
+        ret.append(types.int32)
+        ret.append(types.int32 if is_input_args else types.CPointer(types.int32))
+    return ret
+
+def setup_fn_sig(num_outputs, num_inputs, output_dtypes, input_dtypes):
+    assert num_outputs == len(output_dtypes)
+    assert num_inputs == len(input_dtypes)
+    
+    args_list = _populate_setup_args(output_dtypes)
+    args_list += _populate_setup_args(input_dtypes, True)
+    for i in range(3):
```

Comment:
```suggestion
    for _ in range(3):
```
nitpick: no need for the named variable